scipy/optimize/tests/test_minpack.py

```diff
@@ -517,6 +517,26 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_covariance(self):
```

Comment:
What I meant is that it's a non-trivial part and it must be tested. The doc is [here]( http://scipy.github.io/devdocs/generated/scipy.linalg.cholesky.html#scipy.linalg.cholesky), by default `A = U.T.dot(U)` where `U` is upper-triangular. But we do matrix inversion, so...

Basically it must be tested to have a confidence. By the way I see that `solve_triangular` is used differently in `_wrap_func` and `_wrap_jac` currently.